packages/react-instantsearch-hooks/src/components/__tests__/InstantSearchSSRProvider.test.tsx

```diff
@@ -539,10 +543,15 @@ describe('InstantSearchSSRProvider', () => {
     const { getByRole } = render(<App />);
     const appleRefinement = getByRole('checkbox', { name: 'Apple 442' });
 
+    expect(searchClient.search).toHaveBeenCalledTimes(0);
+    expect(appleRefinement).not.toBeChecked();
+
     userEvent.click(appleRefinement);
 
     await waitFor(() => {
-      expect(appleRefinement).toBeChecked();
+      expect(searchClient.search).toHaveBeenCalledTimes(1);
+      // possible bug in testing-library, this is checked in real life in the same situation
+      // expect(appleRefinement).toBeChecked();
```

Comment:
I don't understand why the bottom assertion fails. It already is the case before this PR actually, as the assertion was written wrong (before this PR it already was checked, now the local state is trusted, and thus not checked).

Because the assertion was originally written wrong, I'm kind of doubting whether https://github.com/algolia/react-instantsearch/issues/3530 was even asserted by this test.

Reply:
This test was never correctly passing (searchClient wasn't correct) and the behaviour changed in this PR. It's sufficiently covered under the expanded "renders refinements from local widget state" test now.